-1

Tested at: 0a13ab858755dcd7ba9bcccd7aa3b4a4edafa806

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-14299/12719/summary.html

I found follow errors while testing this PR

**Failed tests:  AddOn  ClangBuild**

* **AddOn**:

I found errors in the following addon tests:

cmsDriver.py RelVal -s HLT:HIon,RAW2DIGI,L1Reco,RECO --data --scenario=HeavyIons -n 10 --conditions auto:run2_data_HIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2016,Run2_HI --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_HIon_DATA.root --fileout file:RelVal_Raw_HIon_DATA_HLT_RECO.root : FAILED - time: date Fri Apr 29 14:59:32 2016-date Fri Apr 29 14:51:42 2016 s - exit: 34304


* **Clang**:

I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && scram build -k -j 48 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'


 <pre>>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-04-28-2300/src/TrackingTools/TrajectoryCleaning/src/TrajectoryCleanerBySharedSeeds.cc 
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-04-28-2300/src/TrackingTools/TrajectoryCleaning/src/TrajectoryCleanerFactory.cc 
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-04-28-2300/src/TrackingTools/TrajectoryCleaning/src/FastTrajectoryCleaner.cc 
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-04-28-2300/src/TrackingTools/TrajectoryCleaning/src/TrajectoryCleaner.cc 
In file included from /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-04-28-2300/src/TrackingTools/TrajectoryCleaning/src/FastTrajectoryCleaner.cc:1:
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-04-28-2300/src/TrackingTools/TrajectoryCleaning/interface/FastTrajectoryCleaner.h:41:22: error: ISO C++11 does not allow access declarations; use using declarations instead
  TrajectoryCleaner::clean;
                     ^
  using 
1 error generated.
In file included from /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-04-28-2300/src/TrackingTools/TrajectoryCleaning/src/TrajectoryCleanerBySharedSeeds.cc:4:
</pre>

--- a/TrackingTools/TrajectoryCleaning/interface/FastTrajectoryCleaner.h
+++ b/TrackingTools/TrajectoryCleaning/interface/FastTrajectoryCleaner.h
@@ -38,7 +38,6 @@ class FastTrajectoryCleaner final : public TrajectoryCleaner {
 
   ~FastTrajectoryCleaner(){}
 
-  TrajectoryCleaner::clean;
   void clean(TempTrajectoryContainer&) const override;
   void clean(TrajectoryPointerContainer&) const override;
 

--- a/TrackingTools/TrajectoryCleaning/src/FastTrajectoryCleaner.cc
+++ b/TrackingTools/TrajectoryCleaning/src/FastTrajectoryCleaner.cc
@@ -28,7 +28,7 @@ void FastTrajectoryCleaner::clean( TempTrajectoryContainer & tc) const
      maxScore = score;
     }
   }
-  assert(bestTr);
+  
   for (auto & it : tc) {
    if ((&it)!=bestTr) it.invalidate();
   }

